Events.php

```diff
@@ -146,14 +148,43 @@ public static function onTopMenuInit($event)
                     continue;
                 }
 
+                $isCurrentTargetUrl = false;
+                if ($page->type === LinkType::ID && $page->page_content) {
+                    $targetUrl = strpos($page->page_content, 'http') === 0 ?
+                        $page->page_content :
+                        'https://domain.tld/' . trim($page->page_content, '/');
+                    $targetUrlPath = parse_url($targetUrl, PHP_URL_PATH) ?: '';
+                    $targetUrlQuery = parse_url($targetUrl, PHP_URL_QUERY) ?: '';
+                    $container = ContentContainerHelper::getCurrent();
+                    $currentContainerPath = $container ?
+                        rtrim(ContentContainerHelper::getCurrent()->getUrl(), '/') :
+                        null;
+                    if (
+                        $targetUrlPath
+                        && (
+                            $currentContainerPath === $targetUrlPath
+                            || Url::current() === $targetUrlPath . '?' . $targetUrlQuery
+                        )
+                    ) {
+                        $isCurrentTargetUrl = true;
+                    }
+                }
+
                 $event->sender->addItem([
                     'label' => Html::encode(Yii::t('CustomPagesModule.base', $page->title)),
                     'url' => Url::to(['/custom_pages/view', 'id' => $page->id]),
                     'htmlOptions' => ['target' => ($page->in_new_window) ? '_blank' : ''],
                     'icon' => '<i class="fa ' . Html::encode($page->icon) . '"></i>',
-                    'isActive' => (Yii::$app->controller->module
-                        && Yii::$app->controller->module->id === 'custom_pages'
-                        && Yii::$app->controller->id === 'view' && !Yii::$app->controller->contentContainer && Yii::$app->request->get('id') == $page->id),
+                    'isActive' => (
+                        (
+                            Yii::$app->controller->module
+                            && Yii::$app->controller->module->id === 'custom_pages'
+                            && Yii::$app->controller->id === 'view'
```

Comment:
@yurabakhtin Not sure, as you like. For me a new PR ...

Reply:
@marc-farre Thank you for the update, but why don't you add the additional condition `|| Url::to() === $targetUrlPath` what I suggested here https://github.com/humhub/custom-pages/pull/313#discussion_r1418904278 ?
I think the solution will mark a top menu item as active if a link page has an url like `http://humhub.local/people`.

Reply:
@marc-farre Ok, thank you for the explanation, yes I understand why you parse URL before checking.

> > I have tested with a link page http://humhub.local/people and it is not marked as active.
> 
> Do you still have the issue?

![test-people-active-url](https://github.com/humhub/custom-pages/assets/6995524/63884be7-c828-4c57-b734-a7f67adb8586)

Reply:
I see. Thanks for your tests. So the reason was the specific URL rule of the people page: https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/config.php#L17C35-L17C35

And I didn't knew the difference with these 2 methods:
- `Url::to()` => `/people`
- `Url::current()` => `/user/people/index`

Fixed in commit https://github.com/humhub/custom-pages/pull/313/commits/4b5a232499bc645fdca6c3ece9b596563acb8bd5

Reply:
@marc-farre Thank you for the fix, it works for the link page with URL `http://humhub.local/people`.

Please note if the People page is filtered like `http://humhub.local/people?keyword=admin` then such Top Menu Item is not active, I am not sure should we find a solution for this case, if not then you can keep it as is and this PR can be merged.

Reply:
Yes, I have no solutions for added queries with filter forms, pagination or others.
Because in some cases we need to check the query to prevent highlighting a non related page (e.g. an ID in the query).